gcloud/datastore/entity.py

```diff
@@ -77,6 +77,32 @@ def __init__(self, key=None, exclude_from_indexes=()):
         self.key = key
         self._exclude_from_indexes = set(exclude_from_indexes)
 
+    def __eq__(self, other):
+        """Compare two entities for equality.
+
+        Entities compare equal if their keys compare equal, and their
+        properties compare equal.
+
+        :rtype: boolean
+        :returns: True if the entities compare equal, else False.
+        """
+        if not isinstance(other, Entity):
+            raise NotImplementedError
```

Comment:
Ignore my previous comment. I just saw this. Thanks for digging up the link! I am happy to have learned this!